File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        hash_set = set(nums)

        long = 0

        for n in hash_set:
            l = 1

            while n+l in hash_set:
                l += 1
            
            long = max(long, l)
        
        return long


        

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.go
func maxProduct(nums []int) int {
    n := len(nums)

    maxDp := make([]int, n)
    minDp := make([]int, n)

    maxDp[0] = nums[0]
    minDp[0] = nums[0]

    res := nums[0]

    for i := 1; i < n; i++ {
        x := nums[i]

        maxDp[i] = max(
            x,
            maxDp[i-1] * x,
            minDp[i-1] * x,
        )
        minDp[i] = min(
            x,
            maxDp[i-1] * x,
            minDp[i-1] * x,
        )
        res = max(res, maxDp[i])
    }
    return res
}
